ConfigurationSubstitutor/ConfigurationSubstitutor.cs

```diff
@@ -68,7 +68,7 @@ private string ApplySubstitution(IConfiguration configuration, string value, Has
 
                 if (substitutedValue == null && !string.IsNullOrEmpty(_fallbackDefaultValueDelimiter))
                 {
-                    var delimitedVals = capture.Value.Split(new[] { _fallbackDefaultValueDelimiter }, StringSplitOptions.None);
+                    var delimitedVals = capture.Value.Split(new[] { _fallbackDefaultValueDelimiter }, 2, StringSplitOptions.None);
```

Comment:
What's the reason for using 2 here?

Reply:
That's just to make sure that only the first occurrence of the delimiter is used to split the captured value string into exactly two parts. So, in the case where multiple occurrences of the specified delimiter are present, only the first occurrence would be used to tokenize the string. A common example of that is passing URLs as a default value along with `:` as a delimiter; for instance:

```
$(VAR1:http://example.com)
```

There the split should take place at the first occurrence of `:` of the captured value string (i.e., `VAR1:http://example.com`), and return to us the second part fully intact as `http://example.com` and not `http`.